### ISSUE TITLE
[BUG] Fenced multiline code blocks inside lists are broken after switching to CodeMirror back and forth:  {"type":"code","name":"N/A"} 

### ISSUE THREAD

**Issue by makcyd** [mdx-editor/editor#463]
- [+] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [+] I have read the documentation and cannot find an answer.

**Describe the bug**

When using fenced code blocks with tripple backtick inside a list item, switching back and forth to CodeMirror breaks the structure, resulting in an error:

```
Parsing of the following markdown structure failed: {"type":"code","name":"N/A"}.
You can fix the errors in source mode and switch to rich text mode when you are ready.
```

It's not the same as https://github.com/mdx-editor/editor/issues/370, because code block language is set, but indentation and change of structure breaks it.

**To Reproduce**

1. Go to https://mdxeditor.dev/editor/demo
2. Switch to CodeMirror
3. Replace the code content with the following Markdown:

    ````markdown
    * List item one
        ```tsx
        line one
        line two
        ```
    * List item two
    ````
4. Switch to MDX - for now everything is looking good
5. Switch back to CodeMirror - it will show something like:

     ````markdown
      * List item one```tsx
      line one
      line two
      ```
      * List item two
      ````
      Note, the line break and indentation before the fenced code block start got removed
      
6. Now add anything else, e.g. another list item:

    ````markdown
    * List item one```tsx
    line one
    line two
    ```
    * List item two
    * List item three
    ````
7. Switch back to MDX and see the error:

```
Parsing of the following markdown structure failed: {"type":"code","name":"N/A"}.
You can fix the errors in source mode and switch to rich text mode when you are ready.
```

As a result, if you try to edit a markdown document like installation instructions with a step-by-step instructions (lists) with fenced code blocks inside those lists, you'd have to manually add the linebreak before the fence \`\`\` and indent the line for each such code block after each edit, which is quite tiresome.

**Expected behavior**

1. line break and indentation before the fenced block is not removed when switching back-and-forth to CodeMirror
2. code blocks are not broken

**Screenshots**

Short video (0:20) with the reproducer:

https://www.youtube.com/watch?v=6PESBeDQC9c

**Comment by petyosi** [comment]
That's a legit problem, quite tricky to solve, Lexical itself is not happy with blocks nested inside list items. What you're observing is a byproduct [of this line](https://github.com/mdx-editor/editor/blob/7d0d945097f457e6cfaa2324c46a7e863f33cab6/src/plugins/core/MdastParagraphVisitor.ts#L10).

**Comment by makcyd** [comment]
> That's a legit problem, quite tricky to solve, Lexical itself is not happy with blocks nested inside list items. What you're observing is a byproduct [of this line](https://github.com/mdx-editor/editor/blob/7d0d945097f457e6cfaa2324c46a7e863f33cab6/src/plugins/core/MdastParagraphVisitor.ts#L10).

Thanks for the insight, we'll probably try  tackling the issue from our end too, so if we succeed - I'll open a PR :)

PS: Thanks for the project, it's really awesome (and we've tried a lot of alternatives too before we switched to MDX Editor).

**Comment by petyosi** [comment]
It might be a matter of a special casing multiple block elements inside the mdast list item node. The tricky part is that you have an inline node (text) next to a block node (the code block). 

An alternative which might work is fixing this during the lexical -> markdown conversion, this might work, too.

**Comment by cosmoarunn** [comment]
As an alternate solution to this issue is to use Code block editor as described here.
https://mdxeditor.dev/editor/docs/code-blocks
It renders as a text box of code and preview looks good.
The codeblock editor allows to edit the code as well.

**Comment by cosmoarunn** [comment]
![Screenshot 2025-01-03 at 19 32 18](https://github.com/user-attachments/assets/1f1c63fe-78ff-49fd-98a9-94fbc6f8050f)

At least the preview won't crash with codeblck editor!
Neat editor. thanks!

**Comment by makcyd** [comment]
@petyosi could you please have a look at PR https://github.com/mdx-editor/editor/pull/678 when you get a chance? 

Looks like it fixes the issue, so I'd really appreciate having this patch in the official package rather than build custom package out of the fork repo

**Comment by petyosi** [comment]
@makcyd I will test that approach and if it works, I will merge; thanks for chiming in. To confirm, it solves the problem for you?

**Comment by makcyd** [comment]
> [@makcyd](https://github.com/makcyd) I will test that approach and if it works, I will merge; thanks for chiming in. To confirm, it solves the problem for you?

Yes, it does solve the problem, many thanks for merging the PR! Cheers!
